subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6006,8 +6008,15 @@ int hci_iso_handle(struct net_buf *buf, struct net_buf **evt)
 		 * by one additional ISO interval.
 		 */
 		target_event = lll_iso->payload_count / lll_iso->bn;
+		latency_prepare = lll_iso->latency_prepare;
+		if (latency_prepare) {
+			/* big_ref_point has been updated, but payload_count
+			 * hasn't been updated yet - increment target_event to
+			 * compensate
+			 */
+			target_event += latency_prepare;
+		}
```

Comment:
Could you move this up as common so that my other PR () be included in your PR? 
https://github.com/zephyrproject-rtos/zephyr/pull/73583